src/Arriba/Arriba.Server/Middlewares/ArribaNormalizeIdentityMiddleware.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.AspNetCore.Http;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Security.Principal;
+using System.Threading.Tasks;
+
+namespace Arriba.Server.Middlewares
+{
+    public class ArribaNormalizeIdentityMiddleware
+    {
+        private const string ClaimTypeName = "name";
+        private readonly RequestDelegate _next;
+
+        public ArribaNormalizeIdentityMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            if (httpContext.User.Identity.IsAuthenticated)
+            {
+                var user = NormalizeIdentity(httpContext.User);
+                httpContext.User = user;
+            }
+
+            await _next(httpContext);
+        }
+
+        private ClaimsPrincipal NormalizeIdentity(ClaimsPrincipal user)
```

Comment:
Seems like this function could be extracted into another class and then easily tested